- Невозможно использовать с await (т.е. ожидать результата)
- Невозможно определить, когда выполнение завершится (если только не использовать callback)
- Вызывающий код не может перехватить возникающие исключения
Вывод: Везде, где доступно следует использовать async-Task вместо async void!

Подробнее в статье "Why is 'async void' Wrong and What To Do Instead" от fornever.